neuralpp/symbolic/expression.py

```diff
@@ -164,7 +184,7 @@ def new_variable(cls, name: str, type_: ExpressionType) -> Variable:
 
     @classmethod
     @abstractmethod
-    def new_function_application(cls, function: Expression, arguments: List[Expression]) -> FunctionApplication:
+    def new_function_application(cls, function: Expression, arguments: List[Expression]) -> Expression:
```

Comment:
That should be the correct return type, since in some past PR I changed `SymPyExpression.new_function_application()` to necessarily return `FunctionApplication`.

But I'll also need to change the return type of `new_quantifier_expression()`.

Reply:
There's a typo, it should be "to not necessarily return `FunctionApplication`".